apps/librepcb/main.cpp

```diff
@@ -205,7 +205,22 @@ static bool isFileFormatStableOrAcceptUnstable() noexcept {
  ******************************************************************************/
 
 static FilePath determineWorkspacePath() noexcept {
-  FilePath wsPath(Workspace::getMostRecentlyUsedWorkspacePath());
+  static const QString FORCE_VAR_NAME("LIBREPCB_WORKSPACE");
+  FilePath wsPath;
+  QString wsForceStr = qgetenv(FORCE_VAR_NAME.toStdString().c_str());
+
+  if (wsForceStr != "") {
+    wsPath = FilePath(wsForceStr);
+    if (!Workspace::isValidWorkspacePath(wsPath)) {
+      qWarning() << FORCE_VAR_NAME << "set to invalid workspace:" << wsForceStr;
```

Comment:
I wonder if this should actually be replaced (or extended) with a message box. If you don't actively read the console output, you might accidentally work with the wrong workspace since the workspace from the config file will silently be used as a fallback. And even worse, you have no way to abort loading the wrong workspace if the env var contains an invalid path. With a message box (containing a "Cancel" button), you are able to close LibrePCB before your productive v0.1 workspace gets loaded with LibrePCB v0.2 (for example).

Or what do you think about this @Tubbles?

Reply:
Yeah maybe just close LibrePCB? Similar to how opening an already opened workspace closes LibrePCB?

Reply:
I updated the logic, so it now resides directly in the main function (it became so simple by doing that..!). The application now does not load any workspace if the variable is set incorrectly, prints the error dialog, and prints a critical message to stderr.

Reply:
>  Yeah maybe just close LibrePCB? Similar to how opening an already opened workspace closes LibrePCB?

Sounds good to me! :+1:

> I updated the logic, so it now resides directly in the main function (it became so simple by doing that..!).

To be honest, I liked it more when it was in the `determineWorkspacePath()` function :wink: Now there is (a little bit of) duplicate code because `openWorkspace()` is called from two different locations. If the env var detection is located in  `determineWorkspacePath()`, `openWorkspace()` is called only from one location and the overall logic in `main()` stays very simple to understand. Note that to enforce closing LibrePCB, you could simply call `return FilePath();` within `determineWorkspacePath()`, then `main()` will not open any workspace.

Btw, I just had the idea that the error handling with the message box could even be simplified. Actually you could remove the call to `isValidWorkspacePath()` (i.e. not validating the workspace path from the env var), then you'll get a message box automatically since `openWorkspace()` will fail afterwards. Sure, the error message will then be missing the information that the env var caused to load that workspace, but as an alternative you could previously call something like `qInfo() << "Loading workspace from environment variable LIBREPCB_WORKSPACE";` so at least in the terminal we'd see *why* this invalid workspace was opened.

But that's my personal preference (I like to keep code simple :grin:), I'm totally fine to accept your current solution with explicitly opening a message box :slightly_smiling_face:

Reply:
@Tubbles any update on this topic? :slightly_smiling_face:

Reply:
Not much lately, been alot at work 😔

Reply:
OK, no problem :) Just let me know if I can help somehow.

Reply:
@Tubbles With the small refactoring done in #906, this feature has now become even easier to implement (but introduced merge conflicts) - thus I allowed myself to update this PR accordingly :see_no_evil: I hope this is fine for you.

The behavior is now as following: If an invalid workspace path is specified by `LIBREPCB_WORKSPACE`, an error message appears that the workspace could not be opened. After clicking this message away, the workspace chooser dialog appears so the user can immediately choose another workspace - or click "cancel" to close the application.

Reply:
Yes of course 😄🥳 this seems to mesh nicely with #906!

Reply:
Perfect, so let's merge this! :rocket:

Thanks for your contribution! :tada: